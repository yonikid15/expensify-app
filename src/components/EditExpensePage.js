import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense, startRemoveExpense, startEditExpense} from '../actions/expenses';

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };

    onClick = (expense) => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/');
    };

    render() {
        console.log(this.props.state);
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>   
                    </div>
                    <div className="content-container">
                        <ExpenseForm 
                            expense={this.props.expense}
                            onSubmit={this.onSubmit}
                        />
                        <button className="btn btn--secondary" onClick={this.onClick}>
                            Remove Expence
                        </button>
                    </div>
                </div>
            </div>
        );
    }
};

const mapStateToProps = (state, props) => {
    return {
        state: state,
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    };
};

const mapDispatchToProps = (dispatch) => ({ 
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);