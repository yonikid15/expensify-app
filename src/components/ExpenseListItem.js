import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        {
            (id && description && amount) ? (
                <div>
                    <Link to={`/edit/${id}`}>
                        <h3>{description}</h3>
                    </Link>
                    <p>{amount} - {createdAt}</p>
                </div>
            ) : (
                <p>Expense Error</p>
            )
        }
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: state.expenses
    };
};

export default ExpenseListItem;