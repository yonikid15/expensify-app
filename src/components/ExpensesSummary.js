import React from 'react';
import { connect } from 'react-redux';
import getTotalExpenses from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpensesSummary = (props) => (
    <div>
       <p>Viewing {props.expensesCount} expenses totalling {numeral(props.expensesTotal / 100).format('$0,0.00')}</p>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expensesCount: selectExpenses(state.expenses, state.filters).length,
        expensesTotal: getTotalExpenses(selectExpenses(state.expenses, state.filters))
    };
};

export default connect(mapStateToProps)(ExpensesSummary);