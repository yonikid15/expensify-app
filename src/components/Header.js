import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth'; 

export const Header = ({ startLogout }) => (
    <header>
        <h1>Expensify</h1>
        <p><NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink></p>
        <p><NavLink to="/create" activeClassName="is-active">Add Expense</NavLink></p>
        <p><NavLink to="/help" activeClassName="is-active">Help</NavLink></p>
        <button onClick={startLogout}>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);