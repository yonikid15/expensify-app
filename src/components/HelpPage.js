import React from 'react';

const HelpPage = () => (
    <div>
        This is from my help dashboard component.
    </div>
);

export default HelpPage;