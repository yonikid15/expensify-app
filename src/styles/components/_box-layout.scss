.box-layout {
    background: $light-blue;
    background-size: cover;
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
    width: 100vw;
}

.box-layout__box {
    background-color: fade-out($color: white, $amount: .15);
    border-radius: 3px;
    text-align: center;
    width: 25rem;
    padding: $l-size $m-size;
}

.box-layout__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
}