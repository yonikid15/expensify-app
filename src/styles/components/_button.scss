.btn {
    border: none;
    background: $light-blue;
    padding: $m-size $xl-size;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    text-decoration: none;
    line-height: 1;
}

.btn--link {
    background: none;
}

.btn--secondary {
    background: #888888;
}