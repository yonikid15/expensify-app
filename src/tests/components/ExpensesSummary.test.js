import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';
import getTotalExpenses from '../../selectors/expenses-total';
import selectExpenses from '../../selectors/expenses';

test('should render ExpensesSummary with expense count of 1', () => {
    const filters = {
        text: 'credit',
        sortBy: 'amount',
        startDate: undefined,
        endDate: undefined
    };
    const count = selectExpenses(expenses, filters).length;
    const total = getTotalExpenses(selectExpenses(expenses, filters));
    const wrapper = shallow(<ExpensesSummary expensesCount={count} expensesTotal={total}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with expense count of more than 1', () => {
    const filters = {
        text: 'e',
        sortBy: 'amount',
        startDate: undefined,
        endDate: undefined
    };
    const count = selectExpenses(expenses, filters).length;
    const total = getTotalExpenses(selectExpenses(expenses, filters));
    const wrapper = shallow(<ExpensesSummary expensesCount={count} expensesTotal={total}/>);
    expect(wrapper).toMatchSnapshot();
});