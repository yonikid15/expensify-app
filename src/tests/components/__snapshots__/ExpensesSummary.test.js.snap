// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with expense count of 1 1`] = `
<div>
  <p>
    Viewing 
    1
     expenses totalling 
    $45.00
  </p>
</div>
`;

exports[`should render ExpensesSummary with expense count of more than 1 1`] = `
<div>
  <p>
    Viewing 
    2
     expenses totalling 
    $1,140.00
  </p>
</div>
`;
